proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/biometry/impl/BiometricHelper.java

```diff
@@ -134,4 +136,78 @@
             return null;
         }
     }
+
+    /**
+     * Determine if the current device should explicitly fallback {@code FingerprintManager} based
+     * authentication.
+     *
+     * @param context Android context object.
+     * @return {@code true} in case that fallback to {@code FingerprintManager} is required.
+     */
+    public static boolean shouldFallbackToFingerprintManager(@NonNull Context context) {
+        if (Build.VERSION.SDK_INT != Build.VERSION_CODES.P) {
+            return false;
+        }
+        if (!isVendorInFallbackList(context)) {
+            return false;
+        }
+        return isModelInPrefixList(context.getResources().getStringArray(R.array.crypto_fingerprint_fallback_prefixes));
+    }
+
+    /**
+     * Determine if the current device should hide a fingerprint dialog immediately. This is required
+     * for the problematic devices that display it's own, custom fingerprint dialog overlay.
```

Comment:
Also, I would improve the comment a bit to explain what is going on inside, or simplified the conditions.